### PR TITLE
sync RHOSP repos to capsule

### PR DESCRIPTION
**Pull request by vijaysawant**
#### Problem Statement
Closeloop around Bug #2154734 - openstack repos unable to sync on capsule

#### Solution
Added test case in `tests/foreman/api/test_capsulecontent.py` will create & sync rh openstack container repos on satellite and capsule

#### Related Issues
N/A

### CODE REVIEW

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_capsulecontent.py -k test_positive_capsule_sync_openstack_container_repos

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_capsulecontent.py -k test_positive_capsule_sync_openstack_container_repos

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_capsulecontent.py -k test_positive_capsule_sync_openstack_container_repos

**Review by damoore044 — COMMENTED** [review]
Quick question about repo sync-

File: tests/foreman/api/test_capsulecontent.py [hunk 1]
```diff
@@ -1353,3 +1353,71 @@ def test_positive_remove_capsule_orphans(
...
+        for ups_name in upstream_names:
+            repo = entities.Repository(
+                content_type='docker',
+                docker_upstream_name=ups_name,
+                product=function_product,
+                url=constants.RH_CONTAINER_REGISTRY_HUB,
+                upstream_username=settings.subscription.rhn_username,
+                upstream_password=settings.subscription.rhn_password,
+            ).create()
+            repo.sync(timeout=1800)
```
[the remark this answers is not in the document]
Comment (vijaysawant) [reply tests/foreman/api/test_capsulecontent.py:1400]: no required, we can make it to 5 or 10 min for each repo sync

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_capsulecontent.py -k test_positive_capsule_sync_openstack_container_repos

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_capsulecontent.py -k test_positive_capsule_sync_openstack_container_repos

File: tests/foreman/api/test_capsulecontent.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vijaysawant) [reply tests/foreman/api/test_capsulecontent.py]: 10 min for repo sync is not working, incresing to 20 min, lets see again

File: tests/foreman/api/test_capsulecontent.py [hunk 2]
```diff
@@ -1353,3 +1353,71 @@ def test_positive_remove_capsule_orphans(
...
+        function_org,
+        function_product,
+        function_lce,
+    ):
+        """Synchronize openstack container repositories to capsule
+
+        :id: 23e64385-7f34-4ab9-bd63-72306e5a4de0
+
+        :setup:
+            1. A blank external capsule that has not been synced yet with immediate download policy.
```
Comment (vsedmik) [inline tests/foreman/api/test_capsulecontent.py:1370]:
```suggestion
            1. A blank external capsule that has not been synced yet.
```
I couldn't find any requirement for `immediate` download policy in the BZ, nor its usage in the test body. I guess we test with the default `on_demand` policy here.

File: tests/foreman/api/test_capsulecontent.py [hunk 3]
```diff
@@ -1353,3 +1353,71 @@ def test_positive_remove_capsule_orphans(
...
+
+        :steps:
+            1. Enable openstack container repos and sync it to the Library LCE.
```
Comment (vsedmik) [inline tests/foreman/api/test_capsulecontent.py:1373]:
```suggestion
            1. Enable and sync openstack container repos.
            2. Add the repos to a CV, publish and promote it to capsule's LCE.
```

File: tests/foreman/api/test_capsulecontent.py [hunk 4]
```diff
@@ -1353,3 +1353,71 @@ def test_positive_remove_capsule_orphans(
...
+
+        # Create and publish a content view with all repositories
+        cv = entities.ContentView(organization=function_org, repository=repos).create()
+        cv.publish()
+        cv = cv.read()
+        assert len(cv.version) == 1
+
+        # Promote the latest CV version into capsule's LCE
+        cvv = cv.version[-1].read()
+        cvv.promote(data={'environment_ids': function_lce.id, 'force': True})
```
Comment (vsedmik) [inline tests/foreman/api/test_capsulecontent.py:1419]:
```suggestion
        cvv.promote(data={'environment_ids': function_lce.id})
```
IMHO we don't need to use the Force here.

**Review by vsedmik — CHANGES_REQUESTED** [review]
Some minor changes requested, otherwise looks good to me.

File: tests/foreman/api/test_capsulecontent.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vijaysawant) [reply tests/foreman/api/test_capsulecontent.py:1400]: 20 min also not working.. increasing it to 30 min again

File: tests/foreman/api/test_capsulecontent.py (hunk 2 above)
Comment (vijaysawant) [reply tests/foreman/api/test_capsulecontent.py:1370]: Yes, we are not mentioning explicitly which download policy should apply, keeping default

File: tests/foreman/api/test_capsulecontent.py (hunk 4 above)
Comment (vijaysawant) [reply tests/foreman/api/test_capsulecontent.py:1419]: yea, it's working without `'force': True` option
Thanks

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_capsulecontent.py -k test_positive_capsule_sync_openstack_container_repos

**Review by damoore044 — APPROVED** [review]

**Review by vsedmik — APPROVED** [review]

**Comment by vsedmik** [comment]
oh maybe one thing - replace `entities` by `target_sat.api`

**Review by synkd — APPROVED** [review]
ACK, although I agree with @vsedmik that calls using `entities` should be replaced with `target_sat.api`.

**Comment by vijaysawant** [comment]
> ACK, although I agree with @vsedmik that calls using `entities` should be replaced with `target_sat.api`.

Okay. Let me change it quickly

**Comment by vijaysawant** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_capsulecontent.py -k test_positive_capsule_sync_openstack_container_repos

**Pull request merged**